src/plugins/macaddr/macaddr.c

```diff
@@ -0,0 +1,271 @@
+/**
+ * @file
+ *
+ * @brief Source for macaddr plugin
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#include "macaddr.h"
+
+#include <assert.h>
+#include <ctype.h>
+#include <kdberrors.h>
+#include <kdbhelper.h>
+#include <kdbprivate.h>
+#include <regex.h>
+#include <stdio.h>
+
+#define SEPARATORSTANDARD 5
+#define SEPARATOREONE 1
+#define SEPARATORENONE 0
+
+#define MAXMACINT 281474976710655
+
+static void insertSeperator (char * mac)
+{
+	for (int i = 2; i <= 14; i += 3)
+	{
+		mac[i] = ':';
+	}
+	mac[17] = '\0';
+}
+
+void transformMac (Key * key)
+{
+	const char * macKey = keyString (key);
```

Comment:
I am not sure I understand what this function does, but from the way it is used, I inferred that you are trying to transform the mac address from the supported formats into a standard format.

My suggestion would be to do something like this (since you already used a regex to validate the string):
```
output = malloc a 12 character string
outputIndex = 0

foreach char c in input:
    if c is a separator:
        skip
    else:
        output[outputIndex] = c
        outputIndex++
```